service.h

```diff
@@ -107,6 +109,18 @@ class ServiceImpl final : public ServiceLayer::Service {
   std::vector<std::string> parseChirpList(const std::string& childrenlist);
   // parse a list of usernames to individual usernames
   std::vector<std::string> parseUserList(const std::string& userlist);
+
+ public:
+  // Make these member functions public for testing purpose
+  // get the storage client for testing purpose
+  inline StorageClient* const get_storage_client() {
```

Comment:
This is a bit worrysome; I would instead recommend removing this method and make the tests that need access into friends.

Also, inline is usually redundant.